### PR TITLE
feat(lifecycle): `onAction` instead of `onError`

### PR DESCRIPTION
**Pull request by euaaaio**

### CODE REVIEW

**Comment by ai** [comment]
Can I ask you to resolve the conflicts?

@Eddort what do you think about new API?

**Comment by euaaaio** [comment]
> @euaaaio Can I ask you to resolve the conflicts?

Done

**Comment by Eddort** [comment]
I think we need to add test cases to async action. The current realization doesn’t handle async actions. I will do PR later.

**Comment by euaaaio** [comment]
> I think we need to add test cases to async action. The current realization doesn’t handle async actions. I will do PR later.

What do you mean? `onAction` is only about async actions.

**Comment by Eddort** [comment]
When we call two same async actions we can't understand where are correct changes are. It is a race of promises.
I did the test case for it: https://github.com/euaaaio/nanostores/pull/1/files#diff-79c620f45d8436c15063845e42cba347a913694d2baa45263f0e2b7f4cae93e6R303

File: lifecycle/index.js [hunk 1]
```diff
@@ -156,11 +156,42 @@ export let onMount = (store, initialize) => {
...
+
+export let onAction = (store, listener) =>
+  on(store, listener, ACTION, runListeners => {
+    let errorListeners = {}
+    let endListeners = {}
+    let originAction = store.action
     let originError = store.error
-    store.error = (error, actionName) => runListeners({ error, actionName })
+    let originEnd = store.end
+    store.action = (id, actionName, args) => {
```
Comment (ai) [inline lifecycle/index.js:165]: Adding methods to `store` can lead to dangerous things. User extends store object with custom methods/properties.

But it is just a thinking out loud. Anyway, we already started it and should fix it in another PR.

**Pull request merged**